rst_files/generic_programming.rst

```diff
@@ -426,7 +426,7 @@ Numbers and Algebraic Structures
 
 Define two binary functions,  :math:`+` and :math:`\cdot`, called addition and multiplication -- although the operators can be applied to data structures much more abstract than a ``Real``
 
-In mathematics, a `ring <https://en.wikipedia.org/wiki/ring_\(mathematics\)>`_ is a set with associated additive and multiplicative operators where
+In mathematics, a `ring <https://en.wikipedia.org/wiki/ring_\(mathematics\>`_ is a set with associated additive and multiplicative operators where
```

Comment:
Are you sure that this escapes the links properly?  I am not sure if we can check this yet with sphinx, so I suspect it is premature to modifyu it.